src/react/viewport-controller.js

```diff
@@ -87,24 +129,69 @@ export default class ViewportController extends PureComponent {
       onStateChange: this._onInteractiveStateChange.bind(this),
       eventManager: this._eventManager
     }));
+
+    this._transitionManger = new TransitionManager(this.props, this._onTransitionUpdate);
   }
 
   componentWillUpdate(nextProps) {
     if (this._controls) {
       this._controls.setOptions(nextProps);
     }
+    if (this._transitionManger) {
```

Comment:
Shouldn't it read all props, not just viewports?

Reply:
`processViewportChange` takes all props, extracts viewport props from it and processes only a viewport change. In this PR I am not handling `viewports` (multi-viewport prop), that will be in next PR.